File: ecommerce-backend/src/main/java/com/zware/ecommercebackend/dto/Purchase.java
package com.zware.ecommercebackend.dto;

import com.zware.ecommercebackend.entity.Address;
import com.zware.ecommercebackend.entity.Customer;
import com.zware.ecommercebackend.entity.Order;
import com.zware.ecommercebackend.entity.OrderItem;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: angular-ecommerce/src/app/common/Purchase.ts
import {Customer} from "./Customer";
import {Address} from "./Address";
import {Order} from "./Order";
import {OrderItem} from "./OrderItem";

export class Purchase{
  customer: Customer;
  shippingAddress: Address;
  billingAddress: Address;
  order: Order;
  orderItems: OrderItem[];
}
